researchapp/views/consent/views.py

```diff
@@ -0,0 +1,84 @@
+''' Define the routes in consent blueprint.
+'''
+from flask import (
+    Blueprint,
+    jsonify,
+    redirect,
+    render_template,
+    request,
+    session,
+    url_for,
+)
+
+from researchapp.extensions import sync
+
+BP = Blueprint('consent', __name__, template_folder='templates')
+PARTICIPANT = 1
+
+
+@BP.route('/')
+def index():
+    ''' Show homepage.
+    '''
+    return render_template('index.jinja2')
+
+
+@BP.route('/share-my-data')
+def share_my_data():
+    ''' Show "share my data" screen.
+    '''
+    return render_template('share_my_data.jinja2')
+
+
+@BP.route('/consent')
+def consent():
+    ''' Show "consent" screen.
+    '''
+    provider_id = request.args.get('doctor')
+    providers = [provider for provider in sync.list_providers()
+                 if str(provider['id']) == str(provider_id)]
+
+    try:
+        session['provider'] = providers[0]
+        return render_template('consent.jinja2', provider=providers[0])
+    except IndexError:
+        return render_template('invalid_provider.jinja2')
+
+
+@BP.route('/authorize', methods=['POST'])
+def authorize():
+    ''' Start the OAuth process.
+    '''
+    try:
+        provider = session['provider']
```

Comment:
There's also a form field called `provider` as part of the submitted data. Is there a reason we're tracking in both places? What if they disagree? Overall, it would be good to avoid relying on session state for this (the question being: can we ensure this works if a user has two tabs open at once, to complete two authorization processes in parallel)?